### PR TITLE
Create release process documentation

### PR DESCRIPTION
**Pull request by trstringer**
**Description**:

It is important that as an open source software project we have clarity
and transparency on how releases are handled. This includes the
frequency as well as how the community can request for changes. This
also goes into depth about the lifecycle of a request for change and how
it makes it into a release.

Signed-off-by: Thomas Stringer <thomas@trstringer.com>

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| Documentation              | [X] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project? No.

2. Is this a breaking change? No.

3. Has documentation corresponding to this change been updated in the [osm-docs](https://github.com/openservicemesh/osm-docs) repo (if applicable)? Not applicable.

### CODE REVIEW

File: docs/release_process.md
Comment (steeling) [inline docs/release_process.md:15]: How are longer projects handled, ie: cert rotation? These are not yet designed, not yet estimated projects. Even when they are estimated larger projects often tend to slip past deadlines. Do we assign these to a release after they are complete? When they are near completion?

One thing I've seen work is to use an additional label, "release-blocker",  and then we can be more aggressive about adding items to releases.

File: docs/release_process.md
Comment (steeling) [inline docs/release_process.md:13]: I think it makes sense to have at least 2-4 future release milestones in existence at once, as opposed to one at a time. It provides more clarity for users, and allows for better planning.

Comment (steeling) [inline docs/release_process.md:13]: I'd be interested in discussing this prioritization process. Do we have any documentation around that?

File: docs/release_process.md
Comment (trstringer) [reply docs/release_process.md:15]: That's a good question and something I should include in this documentation. For longer projects they should likely have their own milestone that isn't tied to a release. They should be worked in parallel to the release(s) during that time but considered for "next release" when the feature milestone completion date lines up with a release milestone completion date.

Comment (steeling) [reply docs/release_process.md:15]: Otherwise folks will be having tasks in the  *in progress* column that are not attributed to a release.

File: docs/release_process.md
Comment (trstringer) [reply docs/release_process.md:13]: What would be the differences between the multiple future release milestones? What's the benefit in splitting up approved work that isn't prioritized yet?

Comment (trstringer) [reply docs/release_process.md:13]: The prioritization process is a combination of roadmap influence as well as roundtable discussion with the maintainers and the community.

Comment (steeling) [reply docs/release_process.md:13]: it allows us to prioritize and plan past a 3 month horizon, while giving users some clarity on what's to come. 

I think it makes the most sense when we add larger work items (from comment above), to a release, but maybe with a "release-candidate" label, meaning its not a blocker.

Comment (steeling) [reply docs/release_process.md:13]: if any of those conversations/roadmap are public, it may be worth including links, dates, etc in this doc

File: docs/release_process.md
Comment (nojnhuh) [inline docs/release_process.md:5]: The release guide has a section regarding release candidates that I think would fit better in this doc: https://github.com/openservicemesh/osm/blob/edd9ea76246f8f9f673de2db30be1eadef232a18/docs/release_guide.md?plain=1#L13-L19

Then we'd have a clear line between **when** to cut a release (this doc) vs. **how** (the release guide).

Comment (trstringer) [reply docs/release_process.md:5]: Just be clear, @nojnhuh, you're recommending that we move that `Release Candidates` section from `release_guide.md` to `release_process.md` (the new file in this PR)?

Comment (nojnhuh) [reply docs/release_process.md:5]: Yeah. We could of course change the language to make it more consistent with the rest of this doc, but I think that information would fit better here than in the release guide.

Comment (trstringer) [reply docs/release_process.md:5]: Agreed. Done!

File: docs/release_process.md
Comment (trstringer) [reply docs/release_process.md:13]: I think that extra granularity doesn't add enough value to compensate the additional overhead of maintaining that. But perhaps in the future if we feel like that is a gap then we should modify accordingly. I think it is too much on the first pass of this without really having a clear need of that. But I'll defer to other maintainers of the project. // @shashankram @snehachhabria @nojnhuh @jaellio

Comment (shashankram) [reply docs/release_process.md:13]: +1 @trstringer.

**Review by shashankram — APPROVED** [review]
@trstringer thanks for drafting this!

File: docs/release_process.md
Comment (nojnhuh) [reply docs/release_process.md:13]: I don't think we need to always have a set number of upcoming version milestones. We can always have at least one and if we think it would be worth it to have more we can create them as needed. I suspect committing to always having 3 or 4 would be a pain, but giving ourselves a chance to do some extra planning further out wouldn't hurt.

File: docs/release_process.md
Comment (nojnhuh) [inline docs/release_process.md:27]: @trstringer Could you also remove this from the release guide? I don't think we need it in both places.

Comment (trstringer) [reply docs/release_process.md:27]: Good catch, that was an oversight. Removed from the release guide.

**Review by snehachhabria — APPROVED** [review]

**Pull request merged**
